sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,271 @@
+    public void dispose() {
+        stopAudioStream();
+
+        streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+    }
+
+    /**
+     * Starts the audio service and audio stream to the connected device.
+     * The method is non-blocking.
+     * @param encrypted Specify whether or not the audio stream should be encrypted.
+     */
+    public void startAudioStream(boolean encrypted) {
+        // audio stream cannot be started without a connected internal interface
+        if (internalInterface == null || !internalInterface.isConnected()) {
+            return;
+        }
+
+        // streaming state must be NONE (starting the service is ready. starting stream is started)
+        if (streamingStateMachine.getState() != StreamingStateMachine.NONE) {
+            return;
+        }
+
+        try {
+            AudioPassThruCapabilities capabilities = (AudioPassThruCapabilities)internalInterface.getCapability(SystemCapabilityType.PCM_STREAMING);
+
+            switch (capabilities.getSamplingRate()) {
+                case _8KHZ:
+                    sdlSampleRate = 8000;
+                    break;
+                case _16KHZ:
+                    sdlSampleRate = 16000;
+                    break;
+                case _22KHZ:
+                    // common sample rate is 22050, not 22000
+                    // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                    sdlSampleRate = 22050;
+                    break;
+                case _44KHZ:
+                    // 2x 22050 is 44100
+                    // see https://en.wikipedia.org/wiki/Sampling_(signal_processing)#Audio_sampling
+                    sdlSampleRate = 44100;
+                    break;
+            }
+
+            switch (capabilities.getBitsPerSample()) {
+                case _8_BIT:
+                    sdlSampleType = SampleType.UNSIGNED_8_BIT;
+                    break;
+                case _16_BIT:
+                    sdlSampleType = SampleType.SIGNED_16_BIT;
+                    break;
+            }
+
+            streamingStateMachine.transitionToState(StreamingStateMachine.READY);
+
+            internalInterface.addServiceListener(SessionType.PCM, this.serviceListener);
+            internalInterface.startAudioService(encrypted);
+        } catch (Exception e) {
+            Log.e(TAG, "error starting audio stream", e);
+            streamingStateMachine.transitionToState(StreamingStateMachine.ERROR);
+        }
+    }
+
+    /**
+     * Stops the audio service and audio stream to the connected device.
+     * The method is non-blocking.
+     */
+    public void stopAudioStream() {
+        if (internalInterface == null || !internalInterface.isConnected()) {
+            return;
+        }
+
+        // streaming state must be STARTED (starting the service is ready. starting stream is started)
+        if (streamingStateMachine.getState() != StreamingStateMachine.STARTED) {

```

Comment:
I believe this logic is already built into the streamingStateMachine. It will only change state on a valid transition (STARTED->STOPPED) so this check isn't necessary